packages/workbox-precaching/src/PrecacheController.ts

```diff
@@ -310,58 +311,94 @@ class PrecacheController {
   }
 
   /**
-   * Returns a function that looks up `url` in the precache (taking into
-   * account revision information), and returns the corresponding `Response`.
+   * This acts as a drop-in replacement for [`cache.match()`](https://developer.mozilla.org/en-US/docs/Web/API/Cache/match)
+   * with the following differences:
+   * 
+   * - It knows what the name of the precache is, and only checks in that cache.
+   * - It allows you to pass in an "original" URL without versioning parameters,
+   * and it will automatically look up the correct cache key for the currently
+   * active revision of that URL.
+   * 
+   * E.g., `matchPrecache('index.html')` will find the correct precached
+   * response for the currently active service worker, even if the actual cache
+   * key is `'/index.html?__WB_REVISION__=1234abcd'`.
    *
-   * If for an unexpected reason there is a cache miss when looking up `url`,
-   * this will fall back to retrieving the `Response` via `fetch()`.
-   *
-   * @param {string} url The precached URL which will be used to lookup the
-   * `Response`.
-   * @return {workbox.routing.Route~handlerCallback}
+   * @param {string|Request} request The key (without revisioning parameters)
+   * to look up in the precache.
+   * @return {Promise<Response|undefined>}
    */
-  createHandlerForURL(url: string): RouteHandlerCallback {
-    if (process.env.NODE_ENV !== 'production') {
-      assert!.isType(url, 'string', {
-        moduleName: 'workbox-precaching',
-        funcName: 'createHandlerForURL',
-        paramName: 'url',
-      });
-    }
-
+  async matchPrecache(request: string|Request): Promise<Response|undefined> {
+    const url = request instanceof Request ? request.url : request;
     const cacheKey = this.getCacheKeyForURL(url);
-    if (!cacheKey) {
-      throw new WorkboxError('non-precached-url', {url});
+    if (cacheKey) {
+      const cache = await caches.open(this._cacheName);
+      return cache.match(cacheKey);
     }
+    return undefined;
+  }
 
-    return async () => {
+  /**
+   * Returns a function that can be used within a {@link workbox.routing.Route}
+   * that will find a response for the incoming request against the precache.
+   *
+   * If for an unexpected reason there is a cache miss for the request,
+   * this will fall back to retrieving the `Response` via `fetch()` when
+   * `fallbackToNetwork` is `true`.
+   *
+   * @param {boolean} [fallbackToNetwork=true] Whether to attempt to get the
+   * response from the network if there's a precache miss.
+   * @return {workbox.routing.Route~handlerCallback}
+   */
+  createHandler(fallbackToNetwork = true): RouteHandlerCallback {
+    return async ({request}: RouteHandlerCallbackOptions) => {
```

Comment:
Doesn't this also need to take an `event` and call `event.waitUntil()`?

Reply:
When used by our `Router` class, the promise that's returned from a handler will be passed to `event.respondWith()`, and by virtue of that, the service worker will be kept alive until the promise is fulfilled:
 https://github.com/GoogleChrome/workbox/blob/f23924a988a3e321dbcd9e3781f07b1de5dce916/packages/workbox-routing/src/Router.ts#L76-L79

If you're using this handler outside of the `Router` class, then I think it's up to you to figure out whether or not you need to explicitly keep the service worker alive.

Also note that `RouteHandlerCallbackOptions` marks `event` as being optional: https://github.com/GoogleChrome/workbox/blob/f23924a988a3e321dbcd9e3781f07b1de5dce916/packages/workbox-core/src/types.ts#L39-L44

So if it *had* to take an event, we would have to use a type other than `RouteHandlerCallbackOptions` for the parameter.